src/tests/system/tests/test_gdm_passkey.py

```diff
@@ -0,0 +1,441 @@
+    client.vfido.reset()
+    client.vfido.pin_disable()
+    client.vfido.start()
+
+    # Add IPA User
+    ipa.user(testuser).add()
+
+    # Set user_auth_type to passkey
+    ipa.user(testuser).modify(user_auth_type="passkey")
+
+    # Allow authentication without PIN for IPA users
+    ipa.host.conn.run("ipa passkeyconfig-mod --require-user-verification=False", raise_on_error=False)
+
+    # Register passkey with IPA User
+    ipa.user(testuser).passkey_add_register(client=client, pin=None, virt_type="vfido")
+
+    # Login through GDM without PIN
+    client.gdm.click_on("listed?")
+    client.gdm.kb_write(testuser)
+    client.gdm.kb_send("tab")
+    client.gdm.click_on("Security key PIN")
+    client.gdm.kb_send("enter")
+    client.gdm.kb_send("tab")
+    client.gdm.assert_text("Touch security key")
+    client.vfido.touch()
+    time.sleep(5)
+    assert client.gdm.check_home_screen(), "User unable to login or see home screen"
+
+    # Make sure verification is set to true
+    ipa.host.conn.run("ipa passkeyconfig-mod --require-user-verification=True", raise_on_error=False)
+
+
+@pytest.mark.builtwith(client="gdm")
+@pytest.mark.builtwith(client="vfido")
+@pytest.mark.topology(KnownTopology.BareIPA)
+def test_gdm__passkey_login_with_password(client: Client, ipa: IPA):
+    """
+    :title: Login via GDM with password as user with passkey registered
+
+    :setup:
+        1. Configure SSSD for gdm-switchable-auth and pam_cert_auth
+        2. Start SSSD
+        3. Start virtual passkey service
+        4. Add user to IPA and set auth_type to passkey and password
+        6. Register passkey with IPA user
+    :steps:
+        1. Login through GDM using Password
+    :expectedresults:
+        1. Login successful and user sees home screen
+    :customerscenario: False
+    """
+    testuser = "pkgdmuser1"
+    pin = "123456"
+    password = "Secret123"
+
+    # Configure SSSD
+    client.authselect.select("sssd", ["with-mkhomedir"])
+    client.sssd.import_domain("ipa.test", ipa)
+    client.sssd.pam["pam_json_services"] = "gdm-switchable-auth"
+    client.sssd.pam["pam_p11_allowed_services"] = "+gdm-switchable-auth"
+    client.sssd.pam["pam_cert_auth"] = "True"
+    client.sssd.start()
+
+    # Start virtual passkey service
+    client.vfido.reset()
+    client.vfido.pin_set(pin)
+    client.vfido.pin_enable()
+    client.vfido.start()
+
+    # Add IPA User
+    ipa.user(testuser).add(password=password)
+
+    # Set user_auth_type to passkey
+    ipa.user(testuser).modify(user_auth_type=["passkey", "password"])
+
+    # Register passkey with IPA User
+    ipa.user(testuser).passkey_add_register(client=client, pin=pin, virt_type="vfido")
+
+    # Change login from passkey to password and login
+    client.gdm.click_on("listed?")
+    client.gdm.kb_write(testuser)
+    client.gdm.kb_send("tab")
+    client.gdm.kb_send("enter")
+    client.gdm.click_on("Password")
+    client.gdm.click_on("Password")
```

Comment:
Is it duplicate `client.gdm.click_on("Password")` or if not why we need two times?